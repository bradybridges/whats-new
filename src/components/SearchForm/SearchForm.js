import React from 'react';
import './SearchForm.scss';

const SearchForm = (props) => {
  return (
    <form id="search-form">
      <input onChange={event => props.updateNews(event)} type="text" name="search-input" placeholder="Search"/>
    </form>
  )
}

export default SearchForm;