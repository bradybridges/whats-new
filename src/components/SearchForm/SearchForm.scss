$primary-color: #171618fa;
$secondary-color: #3f4446;
$tertiary-color: #e3eef0;

form {
  display: flex;
  margin-right: 32px;
}

input {
  background-color: $tertiary-color;
  border-radius: 4px 0px 0px 4px;
  border: solid 2px #779c43;
  border-right: 0;
  color: #171618fa;
  font-size: 1.5em;
  height: 50px;
  text-align: center;
  width: 300px;
}

#search-btn {
  background-color: $secondary-color;
  border: solid 2px #779c43;
  border-left: 0;
  border-radius: 0px 4px 4px 0px;
  color: $tertiary-color;
  font-size: 1em;
  height: 50px;
  transition: .2s;
  width: 100px;
  &:hover {
    background-color: $primary-color;
    color: $tertiary-color;
    cursor: pointer;
    font-size: 1.2em;
  }
}