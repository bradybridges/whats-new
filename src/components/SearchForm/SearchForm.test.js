import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm', () => {
  let wrapper;
  const updateNewsMock = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<SearchForm 
      updateNews={updateNewsMock}
    />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call updateNews on change of input', () => {
    wrapper.find('input').simulate('change');
    expect(updateNewsMock).toHaveBeenCalled();
  });
})