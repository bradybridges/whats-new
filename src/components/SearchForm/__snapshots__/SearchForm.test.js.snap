// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot 1`] = `
<form
  id="search-form"
>
  <input
    name="search-input"
    onChange={[Function]}
    placeholder="Search"
    type="text"
  />
</form>
`;
